inspirehep/utils/tickets.py

```diff
@@ -214,34 +221,75 @@ def get_queues():
 
     :rtype: dict - with ``name (string)``, ``id (integer)`` properties
     """
-    queues = current_cache.get('rt_queues')
-    if queues:
-        return queues
-    else:
-        queues = _get_all_of("queue")
-        if queues:
-            current_cache.set('rt_queues', queues, timeout=current_app.config.get(
-                'RT_QUEUES_CACHE_TIMEOUT', 86400))
-        return queues
+    return _get_from_cache('rt_queues', partial(_get_all_of, 'queue'))
 
 
 def get_users():
     """Returns list of all users as {id, name} dict
 
     :rtype: dict - with ``name (string)``, ``id (integer)`` properties
     """
-    queues = current_cache.get('rt_users')
-    if queues:
-        return queues
-    else:
-        queues = _get_all_of("user")
-        if queues:
-            current_cache.set('rt_users', queues, timeout=current_app.config.get(
-                'RT_USERS_CACHE_TIMEOUT', 86400))
-        return queues
+    return _get_from_cache('rt_users', partial(_get_all_of, 'user'))
+
+
+def get_rt_user_by_email(email, force_update=False):
+    """Find user info for specified e-mail in RT.
+
+    Args:
+        email(str): Email of user to look for.
+
+    Returns:
+        dict: dictionary of the user {"id":"user/<id>", "Name":"<name>", "EmailAddress": "<e-mail>"}.
+
+    """
+    def _parse_users_with_emails(lines):
+        data = zip(lines[0::3], lines[1::3], lines[2::3])
```

Comment:
nice API :grimacing:

Reply:
You cannot do much more with this beautiful response from RT :P